tests/dummy/app/controllers/logging-example/application-log.js

```diff
@@ -0,0 +1,375 @@
+  /**
+  Flexberry Logging Service
+
+   @property flexberryLoggingService
+   @type String[]
+   */
+  flexberryLoggingService: Ember.inject.service('flexberry-logging'),
+
+  _messageNumber: 0,
+
+  logLevel: 0,
+
+  /**
+    Available test application level settings
+
+    @property settings
+    @type String[]
+   */
+  settings: [],
+
+  /**
+   Default choise in settings
+
+   @property text
+   @type String[]
+   */
+  text: '',
+
+  /**
+   List buttons describers for Supported actions
+   */
+  customButtons:  [{
+      buttonName: 'Assert',
+      buttonAction: 'assertAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: 'Error',
+      buttonAction: 'errorAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: 'Throw',
+      buttonAction: 'throwAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: 'Warn',
+      buttonAction: 'warnAction',
+      buttonClasses: 'ui teal button'
+
+    }, {
+      buttonName: 'Log',
+      buttonAction: 'logAction',
+      buttonClasses: 'ui teal button'
+    }, {
+      buttonName: 'Info',
+      buttonAction: 'infoAction',
+      buttonClasses: 'ui teal button'
+    }, {
+      buttonName: 'Debug',
+      buttonAction: 'debugAction',
+      buttonClasses: 'ui yellow button'
+
+    }, {
+      buttonName: 'Deprecation',
+      buttonAction: 'deprecationAction',
+      buttonClasses: 'ui yellow button'
+    }],
+
+  _router: undefined,
+
+  /**
+   *    Selected jQuery object, containing HTML of confirm modal dialog.
+   *
+   *    @property confirmModalDialog
+   *    @type Object
+   *    @default null
+   */
+  confirmModalDialog: undefined,
+
+  /**
+   *    Content to be displayed in confirm modal dialog.
+   *    It will be displayed only if some confirm needs.
+   *
+   *    @property confirmModalDialogContent
+   *    @type String
+   *    @default 't('components.flexberry-file.confirm-dialog-content')'
+   */
+  confirmModalDialogContent: '',
+
+  /**
+   *    Category to be passed in action confirm modal dialog.
+   *
+   *    @property confirmModalDialogCategory
+   *    @type String
+   *    @default ''
+   */
+  confirmModalDialogCategory: '',
+
+  init() {
+    this._router = getOwner(this).lookup('router:main');
+    this.logLevel = this.get('flexberryLoggingService').flexberryLogLevel;
+    let enumsLoglevel = this.get('flexberryLoggingService').enumsLoglevel;
+    this.settings[0] = '0: OFF';
+    for (let level = 1; level < enumsLoglevel.length; level++) {
+      this.settings[level] = level + ': ' + enumsLoglevel.slice(1, level + 1).join('s, ') + 's';
+    }
+
+    this.text = this.settings[this.logLevel];
+  },
+
+  /**
+    Supported actions
+ */
+  actions: {
+    /**
+     j hghhjl*k
+
+     */
+    setLogLevel(choosed) {
+      this.logLevel = parseInt(choosed.substr(0, 1));
+      this.get('flexberryLoggingService').flexberryLogLevel = this.logLevel;
+    },
+
+    /**
+jhghhjlk
```

Comment:
It's Anastasia's inheritance :-)
I don'nt think do you generate complete analysis :-)
In any case, thanks for the detailed analysis